crates/lace-bin/src/typechecking/test/mod.rs

```diff
@@ -0,0 +1,88 @@
+use crate::error::ResultExt;
+use crate::typechecking::error::TypeError;
+
+macro_rules! parse {
```

Comment:
Weren't we going to do this with closures?

Reply:
yes, I forgot lol. Shall I change that?